TarponSimulator2017/Core/FishFactory.cs

```diff
@@ -1,12 +1,59 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace Tarpon.Core
 {
 	public class FishFactory
 	{
-		public FishFactory ()
+		private static FishFactory UniqueInstance = null;
+
+		/// <summary>
+		/// Kind of Singleton
+		/// </summary>
+		/// <value>The instance.</value>
+		public static FishFactory Instance {
+			get {
+				if (UniqueInstance == null) {
+					UniqueInstance = new FishFactory ();
+					return UniqueInstance;
+				}
+				return UniqueInstance;
+			}
+		}
+
+		private Random rnd = new Random ();
+
+		private FishFactory ()
 		{
 		}
+
+		/// <summary>
+		/// Creates the random fish.
+		/// </summary>
+		/// <returns>The random fish.</returns>
+		public Fish CreateRandomFish ()
+		{
+			int StartAbscisse = rnd.Next (0, 1000);
```

Comment:
A bit hardcoded but should be enough for now, until we have camera. I know, I know...